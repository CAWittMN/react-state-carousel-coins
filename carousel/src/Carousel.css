.Carousel-main {
  display: flex;
  align-items: center;
  justify-content: center;
}

.Carousel-main > i {
  cursor: pointer;
  margin: 10px;
  color: #aaa;
}

.Carousel-main > i:hover {
  color: black;
}

.disabled {
  opacity: 0.25;
  pointer-events: none;
}
