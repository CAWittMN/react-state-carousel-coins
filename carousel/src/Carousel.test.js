import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";

// smoke test

it("renders without crashing", function () {
  render(<Carousel />);
});

// snapshot test

it("matches snapshot", function () {
  const { asFragment } = render(<Carousel />);
  expect(asFragment()).toMatchSnapshot();
});

// props snapshot test

it("matches snapshot with props", function () {
  const { asFragment } = render(
    <Carousel
      cardData={[
        {
          src: "https://unsplash.com/photos/6B1v6Gp3xgQ",
          caption: "Photo by Richard Pasquarella on Unsplash",
        },
        {
          src: "https://unsplash.com/photos/6B1v6Gp3xgQ",
          caption: "Photo by Pratik Patel on Unsplash",
        },
        {
          src: "https://unsplash.com/photos/6B1v6Gp3xgQ",
          caption: "Photo by Josh Post on Unsplash",
        },
      ]}
      title="Shells from far away beaches."
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

// end-to-end test

it("works when you click on the right arrow", function () {
  const { queryByTestId, queryByAltText } = render(<Carousel />);

  // expect the first image to show, but not the second
  expect(
    queryByAltText("Photo by Richard Pasquarella on Unsplash")
  ).toBeInTheDocument();
  expect(
    queryByAltText("Photo by Pratik Patel on Unsplash")
  ).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = queryByTestId("right-arrow");
  fireEvent.click(rightArrow);

  // expect the second image to show, but not the first
  expect(
    queryByAltText("Photo by Richard Pasquarella on Unsplash")
  ).not.toBeInTheDocument();
  expect(
    queryByAltText("Photo by Pratik Patel on Unsplash")
  ).toBeInTheDocument();
});

it("works when you click on the left arrow", function () {
  const { queryByTestId, queryByAltText } = render(<Carousel />);
  const rightArrow = queryByTestId("right-arrow");
  fireEvent.click(rightArrow);
  const leftArrow = queryByTestId("left-arrow");
  fireEvent.click(leftArrow);
  expect(
    queryByAltText("Photo by Richard Pasquarella on Unsplash")
  ).toBeInTheDocument();
  expect(
    queryByAltText("Photo by Pratik Patel on Unsplash")
  ).not.toBeInTheDocument();
});

it("hides left arrow on first image", function () {
  const { queryByTestId } = render(<Carousel />);
  const leftArrow = queryByTestId("left-arrow");
  expect(leftArrow).toHaveClass("disabled");
});

it("hides right arrow on last image", function () {
  const { queryByTestId } = render(<Carousel />);
  const rightArrow = queryByTestId("right-arrow");
  expect(rightArrow).not.toHaveClass("disabled");
  fireEvent.click(rightArrow);
  expect(rightArrow).not.toHaveClass("disabled");
  fireEvent.click(rightArrow);
  expect(rightArrow).toHaveClass("disabled");
});
