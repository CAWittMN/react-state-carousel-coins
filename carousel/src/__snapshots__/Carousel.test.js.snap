// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Carousel"
  >
    <h1>
      Shells from far away beaches.
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="fas fa-chevron-circle-left fa-2x disabled"
        data-testid="left-arrow"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          Photo by Richard Pasquarella on Unsplash
        </h4>
        <img
          alt="Photo by Richard Pasquarella on Unsplash"
          class="Card-image"
          src="image1.jpg"
        />
        <small
          class="Card-small"
        >
          Image 1 of 3.
        </small>
      </div>
      <i
        class="fas fa-chevron-circle-right fa-2x "
        data-testid="right-arrow"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot with props 1`] = `
<DocumentFragment>
  <div
    class="Carousel"
  >
    <h1>
      Shells from far away beaches.
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="fas fa-chevron-circle-left fa-2x disabled"
        data-testid="left-arrow"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          Photo by Richard Pasquarella on Unsplash
        </h4>
        <img
          alt="Photo by Richard Pasquarella on Unsplash"
          class="Card-image"
          src="https://unsplash.com/photos/6B1v6Gp3xgQ"
        />
        <small
          class="Card-small"
        >
          Image 1 of 3.
        </small>
      </div>
      <i
        class="fas fa-chevron-circle-right fa-2x "
        data-testid="right-arrow"
      />
    </div>
  </div>
</DocumentFragment>
`;
